cicd/internal/op/maven.go

```diff
@@ -36,8 +43,14 @@ func RunMavenOnPom(pom string, cmd string, args ...string) error {
 // Runs the given Maven command on a specified module. Considering the input, this is equivalent to:
 //
 //	mvn -B {cmd} -f {pom} -pl {module} {args...}
-func RunMavenOnModule(pom string, cmd string, module string, args ...string) error {
-	// fullArgs := []string{"-pl", module}
-	// fullArgs = append(fullArgs, args...)
-	return RunMavenOnPom(pom, cmd, args...)
+func RunMavenOnModule(pom string, cmd string, module []string, args ...string) error {
+	fullArgs := args
+	for _, spanner_module := range SPANNER_MODULES {
```

Comment:
I don't really like this change as it stands because it has the potential to filter out real failures in non-spanner templates. For example, imagine I make a change in 2 templates - one spanner, one non-spanner (this is common in large-scale full-repo changes, or when doing something like regenerating docs as part of a PR) - the tests will not run on the non-spanner template. We also don't seem to have a clear path to fix the underlying problem, so this hack will remain in place indefinitely.

I'd like to propose a different path:

1) Get rid of the whole concept of only building changed files (basically greatly simplify https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/0952eecc097a06ffcc4977927311edfbe5a8e2bc/cicd/internal/workflows/maven-workflows.go#L204)
2) Instead, allow individual workflows to pass in a set of modules to run against, such that you can trigger tests with `./cicd/run-unit-tests --changed-modules="v2/datastream-to-spanner,v2/sourcedb-to-spanner,v2/spanner-change-streams-to-sharded-file-sink"`. If helpful, this could probably even accept aliases (e.g. `--changed-modules="spanner_all"`)
3) Set up a spanner-only GitHub Actions workflow which passes in these modules and only triggers on spanner changes.
4) Exclude spanner from https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/0952eecc097a06ffcc4977927311edfbe5a8e2bc/.github/workflows/java-pr.yml#L24

Thoughts?

Reply:
Having some issues in getting the build to work in isolation - 

The following command is generated - 

```
mvn -B clean install -f pom.xml -e -Dmdep.analyze.skip -Djib.skip -DskipShade -Dspotless.check.skip -Dmaven.test.skip -T3 -pl=v2/datastream-to-spanner/,v2/spanner-common/,v2/spanner-change-streams-to-sharded-file-sink/,v2/gcs-to-sourcedb/,v2/spanner-migrations-sdk/,v2/spanner-custom-shard/,v2/sourcedb-to-spanner/
```

This works locally, however fails on Github actions with - 

https://paste.googleplex.com/6486993152507904

I tried adding the `common` module as a dependency to `datastream-to-spanner` but that does not seem to help. 
Do you have any idea on how to solve this?